Document benefits from TypeScript annotation (#7423)
stylelint/stylelint: docs/developer-guide/plugins.md

  url: "https://github.com/foo-org/stylelint-selector-no-foo/blob/main/README.md"
};

> /** @type {import('stylelint').Rule} */

Reviewer: Should this also demonstrate the generic types that can be declared?

Author: Right, the `Rule` type can receive type parameters for primary and secondary options. But I think simplifying the code example is better here so that people can quickly understand it.

Reviewer: I agree

export default createPlugin(ruleName, ruleFunction);
```

> [!NOTE]

> > The JSDoc annotation (`@type`) benefits auto-completion or type-check by TypeScript.

Reviewer 2: minor

There is something wrong with this sentence.
If I had to guess you mean something like
```md
The JSDoc `@type` tag enables auto-completion and type-checking provided by Typescript.
```

Author: Looks good. Can you suggest code via **Add a suggestion** button? I'll take your suggestion. 😃 

<img width="207" alt="image" src="https://github.com/stylelint/stylelint/assets/473530/dbeaa576-a3b3-40f4-9949-b2b9f56c7fa5">

Reviewer 2: I wanted to know if that what you meant first.

```suggestion
> The [JSDoc `@type` tag](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type) enables the autocompletion and type-checking features provided by Typescript.
```

> > The [JSDoc `@type` tag](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type) enables the autocompletion and type-checking features provided by Typescript.

Reviewer 3: ```suggestion
> The [`@type` JSDoc annotation](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type) enables Typescript to autocomplete and type-check.
```
